### ISSUE TITLE
cargo lints: cargo metadata output leaks into clippy warnings?

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#6508]
I noticed this by checking `syn-1.0.54` downloaded from crates.io.

When I run `cargo clippy -- --cap-lints warn -Wclippy::cargo`, it looks like there is output leaking from cargo-metadata into the clippy warning such as the "warning: could not read ... Downloading crates.." stuff.

````
   Compiling proc-macro2 v1.0.24
    Checking unicode-xid v0.2.1
   Compiling syn v1.0.54 (/home/matthias/vcs/github/rust-clippy/target/crater/crates/syn-1.0.54)
warning: package `syn` is missing `package.keywords` metadata
  |
  = note: `-W clippy::cargo-common-metadata` implied by `-W clippy::cargo`
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#cargo_common_metadata

    Checking quote v1.0.8
warning: could not read cargo metadata: Error during execution of `cargo metadata`:  Downloading crates ...
  Downloaded syn-test-suite v0.0.0
error: failed to verify the checksum of `syn-test-suite v0.0.0`

  |
  = note: `-W clippy::multiple-crate-versions` implied by `-W clippy::cargo`
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#multiple_crate_versions

warning: 2 warnings emitted

warning: package `syn` is missing `package.keywords` metadata
  |
  = note: `-W clippy::cargo-common-metadata` implied by `-W clippy::cargo`
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#cargo_common_metadata

warning: could not read cargo metadata: Error during execution of `cargo metadata`:  Downloading crates ...
  Downloaded syn-test-suite v0.0.0
error: failed to verify the checksum of `syn-test-suite v0.0.0`

  |
  = note: `-W clippy::multiple-crate-versions` implied by `-W clippy::cargo`
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#multiple_crate_versions

warning: 2 warnings emitted

    Finished dev [unoptimized + debuginfo] target(s) in 6.24s
````

clippy @ 015496579dae282a7215a6ad17fe053342d8b825

**Comment by matthiaskrgr** [comment]
Turns this actually printed by clippy:
````rust
#[macro_export]
macro_rules! unwrap_cargo_metadata {
    ($cx: ident, $lint: ident, $deps: expr) => {{
        let mut command = cargo_metadata::MetadataCommand::new();
        if !$deps {
            command.no_deps();
        }

        match command.exec() {
            Ok(metadata) => metadata,
            Err(err) => {
                span_lint($cx, $lint, DUMMY_SP, &format!("could not read cargo metadata: {}", err));
                return;
            },
        }
    }};
}
````
